content/PGP/YubiKey 5.2.3 Enhancements to OpenPGP 3.4.adoc

```diff
@@ -0,0 +1,168 @@
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 19 brainpoolP256r1" /bye
+-----------------
+* curve25519
+[source, python]
+-----------------
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 22 ed25519" /bye
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 2 22 cv25519" /bye
+-----------------
+
+=== Attestation
+YubiKeys offer a new feature to the OpenPGP Smart Card, the attestation of Keys generated on device. With the attestation function, generating an Authentication, Signature or Decipher key on a YubiKey will also create an X.509 certificate signed with the Attestation key present on the device. This certificate can be used to verify any other key generated on the device and was not imported, ensuring said key only exists on the YubiKey. 
+
+Attestation is supported for all combinations of attestation key and attested key, except the attestation key cannot be from the curve25519 family. A fourth key slot is reserved for an attestation key.  If specifying it by ID, it is Key ID 0x81. The Attestation template certificate Is stored on a fourth user certificate slot reserved for it. It is selected directly with SELECT_DATA index 4.  It can also be retrieved by calling GET_NEXT_DATA three times after SELECT_DATA on index 1.
+
+Each YubiKey comes with a pre-loaded Attestation certificate signed by a Yubico OpenPGP CA. Attestation is a Yubikey-proprietary extension. The https://www.google.com/url?q=https://www.yubico.com/products/services-software/download/yubikey-manager/&sa=D&ust=1566260477099000[YubiKey Manager] application can be used to manage the Attestation Certificate. For more details, refer to the OpenPGP Attestation page at: https://developers.yubico.com/PGP/Attestation.html
+
+=== Configuration Enhancements
+Alongside new cryptographic curves and key attestation, the YubiKey 5.2.3 and up supports advanced configuration enhancements, allowing more security and extending functionality for existing features.
+
+=== Key Derived Format
+To remove the transmission and on-card storage of OpenPGP PINs in plain text, the YubiKey supports the Key Derived Function (KDF) functionality. With the KDF function enabled, the PIN is stored as a hash on the YubiKey. When entering the PIN to the OpenPGP Smart Card, the OpenPGP client will only pass the hashed value, never passing the PIN directly. KDF functionality is set on the card itself, and communicated to the client; it is transparent to the user. Should the KDF functionality not be enabled, the PIN function will work as previously. The KDF function is listed in section 4.3.2  of the https://www.google.com/url?q=https://gnupg.org/ftp/specs/OpenPGP-smart-card-application-3.4.pdf&sa=D&ust=1566260477100000[OpenPGP Smart Card 3.4] spec.
+
+Current stable releases (2.2.12) of gnupg support enabling KDF, but do not have any indication when it is enabled. This will be added in the upcoming GnuPG 2.3.0, which displays KDF state as part of the card status.

```

Comment:
I haven't tested, but I believe this is fixed.  I think it's safe to drop this warning regardless.

Either way, these version numbers aren't up-to-date.  2.2.12 is 1.5 years old.  Current version is 2.2.20, and I'm not sure that "2.3.0" is planned for any foreseeable future.